ext/ruby_newt/ruby_newt.c

```diff
@@ -184,28 +184,21 @@ static VALUE rb_ext_ColorSetCustom(VALUE self, VALUE id)
   return INT2NUM(NEWT_COLORSET_CUSTOM(NUM2INT(id)));
 }
 
-static VALUE rb_ext_Screen_new()
+static VALUE rb_ext_Screen_Init()
 {
   if (initialized == Qtrue)
     return Qnil;
 
   newtInit();
-  newtCls();
   memcpy(&newtColors, &newtDefaultColorPalette, sizeof(struct newtColors));
   initialized = Qtrue;
   return Qnil;
 }
 
-static VALUE rb_ext_Screen_Init()
+static VALUE rb_ext_Screen_new()
 {
-  if (initialized == Qtrue)
-    return Qnil;
-
-  newtInit();
-  memcpy(&newtColors, &newtDefaultColorPalette, sizeof(struct newtColors));
-  rb_ext_Widget_Hash = rb_hash_new();
-  rb_gc_register_address(&rb_ext_Widget_Hash);
-  initialized = Qtrue;
+  rb_ext_Screen_Init();
+  newtCls();
   return Qnil;
 }
```

Comment:
When called twice, cls is called twice now, probably expected behavior.

Unsure about rb_gc_register_address change tho, I do not see a change for the unregister counterpart in this commit (probably somewhere else)?